From 145ad72a7f98a9203c3c3118559e034bae1623d7 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Wed, 21 Jul 2021 09:53:47 +0530
Subject: [PATCH 14/21] Add bitset typecasts to make MSVC<v19.20 happy

This patch resolves the following error with the MSVC compiler:

router\src\http\src\http_server_plugin.cc(...): error C2398: Element '1':
conversion from '...' to 'unsigned __int64' requires a narrowing conversion
---
 router/src/http/src/http_server_plugin.cc | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/router/src/http/src/http_server_plugin.cc b/router/src/http/src/http_server_plugin.cc
index eb2c556ca12..7a0fb66d5c5 100644
--- a/router/src/http/src/http_server_plugin.cc
+++ b/router/src/http/src/http_server_plugin.cc
@@ -182,7 +182,7 @@ void HttpRequestThread::set_request_router(HttpRequestRouter &router) {
 }
 
 void HttpRequestThread::wait_and_dispatch() {
-  event_base_.once(-1, {EventFlags::Timeout},
+  event_base_.once(-1, {(EventFlags::Bitset)EventFlags::Timeout},
                    HttpRequestThread::on_event_loop_ready, this, nullptr);
   event_base_.dispatch();
 
@@ -290,7 +290,8 @@ class HttpsRequestMainThread : public HttpRequestMainThread {
         [](EventBase *base, void *arg) {
           return EventBuffer(base, -1, static_cast<TlsServerContext *>(arg),
                              EventBuffer::SslState::Accepting,
-                             {EventBufferOptionsFlags::CloseOnFree});
+                             {(EventBufferOptionsFlags::Bitset)
+                                  EventBufferOptionsFlags::CloseOnFree});
         },
         tls_ctx);
   }
@@ -312,7 +313,8 @@ class HttpsRequestWorkerThread : public HttpRequestWorkerThread {
         [](EventBase *base, void *arg) {
           return EventBuffer(base, -1, static_cast<TlsServerContext *>(arg),
                              EventBuffer::SslState::Accepting,
-                             {EventBufferOptionsFlags::CloseOnFree});
+                             {(EventBufferOptionsFlags::Bitset)
+                                  EventBufferOptionsFlags::CloseOnFree});
         },
         tls_ctx);
   }
-- 
2.41.0

